packages/core/src/components/popover/popover.tsx

```diff
@@ -495,15 +502,15 @@ export class Popover extends AbstractPureComponent2<IPopoverProps, IPopoverState
 
     private handlePopoverClick = (e: React.MouseEvent<HTMLElement>) => {
         const eventTarget = e.target as HTMLElement;
+        const eventPopover = eventTarget.closest(`.${Classes.POPOVER}`);
+        const isEventFromSelf = eventPopover === this.popoverRef.current;
+        const isEventPopoverCapturing = eventPopover.classList.contains(Classes.POPOVER_CAPTURING_DISMISS);
         // an OVERRIDE inside a DISMISS does not dismiss, and a DISMISS inside an OVERRIDE will dismiss.
         const dismissElement = eventTarget.closest(`.${Classes.POPOVER_DISMISS}, .${Classes.POPOVER_DISMISS_OVERRIDE}`);
         const shouldDismiss = dismissElement != null && dismissElement.classList.contains(Classes.POPOVER_DISMISS);
         const isDisabled = eventTarget.closest(`:disabled, .${Classes.DISABLED}`) != null;
-        if (shouldDismiss && !isDisabled && !e.isDefaultPrevented()) {
+        if (shouldDismiss && !isDisabled && (!isEventPopoverCapturing || isEventFromSelf)) {
```

Comment:
is `isDisabled` still relevant here? Not sure if the check was to avoid `preventDefault` on an event that should not have been touched, or if we genuinely didn't want to close the popover if its target is disabled

Reply:
I'm pretty sure it's redundant, since a disabled element can't emit a click event. However I don't know if any blueprint component only uses disabled class but doesn't respect that rule (I checked anchor button and they do respect not triggering onClick) if there's no element which applies disabled class and allow clicking I can remove that too.